app/commands/incident.py

```diff
@@ -361,3 +373,171 @@ def generate_success_modal(body):
     }
+
+
+def handle_reaction_added(client, ack, body, logger):
+    ack()
+    # get the channel in which the reaction was used
+    channel_id = body["event"]["item"]["channel"]
+    # Get the channel name which requires us to use the conversations_info API call
+    channel_name = client.conversations_info(channel=channel_id)["channel"]["name"]
+
+    # if the emoji added is a floppy disk emoji and we are in an incident channel, then add the message to the incident timeline
+    if channel_name.startswith("incident-"):
+        # get the message from the conversation
+        try:
+            result = client.conversations_history(
+                channel=channel_id,
+                limit=1,
+                inclusive=True,
+                include_all_metadata=True,
+                oldest=body["event"]["item"]["ts"],
+            )
+            # get the actual message from the result. This is the text of the message
+            messages = result["messages"]
+
+            # if there are no messages, then the message is part of a thread, so obtain the message from the thread.
+            if messages.__len__() == 0:
+                # get the latest message from the thread
+                result = client.conversations_replies(
+                    channel=channel_id,
+                    ts=body["event"]["item"]["ts"],
+                    inclusive=True,
+                    include_all_metadata=True,
+                )
+                # get the message
+                messages = result["messages"]
+
+            # get the incident report document id from the incident channel
+            # get and update the incident document
+            document_id = ""
+            response = client.bookmarks_list(channel_id=channel_id)
+            if response["ok"]:
+                for item in range(len(response["bookmarks"])):
+                    if response["bookmarks"][item]["title"] == "Incident report":
+                        document_id = extract_google_doc_id(
+                            response["bookmarks"][item]["link"]
+                        )
+                        if document_id == "":
+                            logger.error("No incident document found for this channel.")
+
+            for message in messages:
+                # convert the time which is now in epoch time to standard EST Time
+                message_date_time = convert_epoch_to_datetime_est(message["ts"])
+                # get the user name from the message
+                user = client.users_profile_get(user=message["user"])
+                # get the full name of the user so that we include it into the timeline
+                user_full_name = user["profile"]["real_name"]
+
+                # get the current timeline section content
+                content = get_timeline_section(document_id)
+
+                # if the message already exists in the timeline, then don't put it there again
+                if message_date_time not in content:
```

Comment:
Should we also have a check here to make sure the timeline section was found?
```suggestion
                if content and message_date_time not in content:
```

Reply:
The content is the the current content in the document that is essentially the detailed timeline we have generated so far. By default the content will always be in the content: ie 
content =  get_timeline_section(document_id) return "hello" and lets message date is December 31, 2022 10:56:44
so our statement` if content and message date not in content `would be the equivalent of `if message_date_time not in content` as this is the only thing that decides on the outcome of the if statement. So essentially it will be pointless to make that comparison. Does it make sense? I could be totally misunderstanding it and honestly my brain is a little bit fried so I might not be making any sense here.

Reply:
heh, it is my belt/suspenders nature showing because there is a chance for the `get_timeline_section` function to return `None` if the default template content can't be found:
https://github.com/cds-snc/sre-bot/pull/383/files#diff-5c3a2c3147f439fbafa109f944d12cb934a3316a93905242d47513785b1905e6R327

I'm fine merging without the check because you're totally right, but default that section will be there.

Reply:
You are right that the function could return None I have committed your code suggestion!